components/psst/browser/content/psst_tab_helper_browsertest.cc

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include <string>
+
+#include "base/path_service.h"
+#include "brave/components/constants/brave_paths.h"
+#include "brave/components/psst/browser/core/psst_rule_service.h"
+#include "brave/components/psst/common/features.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/browser/ui/browser_commands.h"
+#include "chrome/test/base/in_process_browser_test.h"
+#include "chrome/test/base/ui_test_utils.h"
+#include "content/public/test/browser_test.h"
+#include "content/public/test/browser_test_utils.h"
+#include "content/public/test/content_mock_cert_verifier.h"
+#include "net/dns/mock_host_resolver.h"
+#include "net/test/embedded_test_server/embedded_test_server.h"
+
+class PsstTabHelperBrowserTest : public InProcessBrowserTest {
```

Comment:
this file can't be in components because it has `chrome/browser` references, but that also doesn't seem right to me because that will prevent this test from working on android. I think also it needs to subclass a different test class to work on desktop and android?

Reply:
you should be using `PlatformBrowserTest` here and also this isn't right https://github.com/brave/brave-core/pull/20336/files#diff-088d05e5f1fc402cbe7970fe0ed8fbd4954e8b8356a28e50cb9887ddd195b816R43
I assume this is failing on android right now because of this

Reply:
this conditional is correct, but you didn't actually use `PlatformBrowserTest` here https://github.com/brave/brave-core/pull/20336/files#diff-088d05e5f1fc402cbe7970fe0ed8fbd4954e8b8356a28e50cb9887ddd195b816R43

but you should also have
```
#if BUILDFLAG(IS_ANDROID)
#include "chrome/test/base/android/android_browser_test.h"
#else
#include "chrome/test/base/in_process_browser_test.h"
#endif
``` 

here to match.